src/vsr/replica.zig

```diff
@@ -1600,6 +1600,8 @@ pub fn ReplicaType(
                 return;
             }
 
+            self.replicate(message);
+
             if (self.status != .normal) {
                 log.debug("{}: on_prepare: ignoring ({})", .{ self.replica, self.status });
                 return;
```

Comment:
Ok, the `(message.header.view < self.view)` condition looks sus to me!

If I am reading `is_repair` correctly ,we are only ending up in that branch when we are recovering head. If we are normal or view change, that we consider this case a repair.

But this means, if we are forwarding this, we either forward it to a recovering head, which will further forward, or we are forwarding to normal or start_view, in which case we are forwarding a repair, which we shouldn't do. 

_And_ in on repair, we already have 

            if (self.status != .normal and self.status != .view_change) {
                log.debug("{}: on_repair: ignoring ({})", .{ self.replica, self.status });
                return;
            }


So perhaps we should just always consider `view < view` case as repair, even in recovering head, and here assert that the view is >=?

If the view is >, that's the case where we _correctly_ should forward a prepare, because we won't be jump viewing immediately, and instead waiting for an SV from the primary

Reply:
Isn't the opposite true for _is_repair_, wherein we are only entering the branch if we are in *normal/view_change*? 

```C
        fn is_repair(self: *const Self, message: *const Message.Prepare) bool {
            assert(message.header.command == .prepare);

            if (self.status == .normal) {
                if (message.header.view < self.view) return true;
                if (message.header.view == self.view and message.header.op <= self.op) return true;
            } else if (self.status == .view_change) {
                if (message.header.view < self.view) return true;
                // The view has already started or is newer.
            }

            return false;
        }
```

Are you talking about a case where we receive a prepare as a *recovering_head* replica? In that case, wouldn't we just be forwarding the prepare without doing anything with it locally? 🤔

Reply:
I mean that in `on_repair` we havndle the case where we neither normal, nor view_change already. So in this sense `on_repair` and `is_repair` are self-inconsistent. I suggest treating `on_repair` as a source of truth, and always rout prepares from older views as repairs, which is a simpler mental model.